certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/action/model/ActionRuleCertificateHasComplement.java

```diff
@@ -11,7 +13,10 @@ public boolean evaluate(Optional<Certificate> certificate,
     return certificate.filter(
         value -> value.messages() != null && !value.messages().isEmpty()
             && value.messages().stream()
-            .anyMatch(message -> message.complements() != null && !message.complements().isEmpty())
+            .anyMatch(message ->
+                message.type() == MessageType.COMPLEMENT
```

Comment:
Could remove the type check and instead use method value.messages(MessageType.COMPLEMENT) to get complements.